### PR TITLE
targets/ramips-mt76x8: add device TP-Link RE200 V3

### PR DESCRIPTION
**Pull request by misanthropos**
Signed-off-by: Richard Fröhning <misanthropos@gmx.de>

- [x] Must be flashable from vendor firmware
  - [x] Web interface
  - [x] TFTP
  - ~~Other: <specify>~~
- [x] Must support upgrade mechanism
  - [x] Must have working sysupgrade
    - [x] Must keep/forget configuration (`sysupgrade [-n]`, `firstboot`)
  - [x] Gluon profile name matches autoupdater image name
        (`lua -e 'print(require("platform_info").get_image_name())'`)
- [x] Reset/WPS/... button must return device into config mode
- [x] Primary MAC address should match address on device label (or packaging)
      (https://gluon.readthedocs.io/en/latest/dev/hardware.html#hardware-support-in-packages)
  - When re-adding a device that was supported by an earlier version of Gluon, a
    factory reset must be performed before checking the primary MAC address, as
    the setting from the old version is not reset otherwise.
- Wired network
  - [x] should support all network ports on the device
  - [x] must have correct port assignment (WAN/LAN)
    - On devices supplied via PoE, there is usually no explicit WAN/LAN labeling on the hardware.
      The PoE input should be the WAN port in this case.
- Wireless network (if applicable)
  - [x] Association with AP must be possible on all radios
  - [x] Association with 802.11s mesh must work on all radios 
  - [x] AP+mesh mode must work in parallel on all radios
- LED mapping
  - Power/system LED
    - [x] Lit while the device is on
    - [x] Should display config mode blink sequence 
          (https://gluon.readthedocs.io/en/latest/features/configmode.html)
  - Radio LEDs
    - [x] Should map to their respective radio
    - [x] Should show activity
  - Switch port LEDs
    - [x] Should map to their respective port (or switch, if only one led present) 
    - [x] Should show link state and activity
- Outdoor devices only:
  - ~~Added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`~~
- Docs:
  - [x] Added Device to `docs/user/supported_devices.rst`

### CODE REVIEW

**Comment by AiyionPrime** [comment]
Closing in favor of the existing #2086 with the very same branch ([misanthropos:re200v3](https://github.com/misanthropos/gluon/tree/re200v3)).

**Comment by AiyionPrime** [comment]
My bad. As @blocktrron already stated in the old PR it was not opened against master, but next.
Anyone missing the history can read up on #2086.

**Review by AiyionPrime — CHANGES_REQUESTED** [review]
@misanthropos from which build are your test results?
Does wifi really work for this revision on both radios without errors?

If nothing else, please add the device to `docs/user/supported_devices.rst`.

**Comment by misanthropos** [comment]
> @misanthropos from which build are your test results? Does wifi really work for this revision on both radios without errors?

Tested with: gluon v2022.1.1 with FFM feature-gluon-2022.1

Both radios are connectable and show according SSIDs including mesh IDs.
Tested 2.4G meshing via phone communicating on 5G with router which has been placed near a FFFM router without attached network cable.
Meshing on 5G must still be tested due to lack of a second test device with 5G capabilities.

**Comment by AiyionPrime** [comment]
Please update the checklist accordingly.

File: docs/user/supported_devices.rst [hunk 1]
```diff
@@ -407,6 +407,7 @@ ramips-mt76x8
   - Archer C50 (v3)
   - Archer C50 (v4)
   - RE200 (v2)
+  - RE200 (v3)
```
Comment (AiyionPrime) [inline docs/user/supported_devices.rst:410]:
```suggestion
  - RE200 (v2, v3)
```

Comment (AiyionPrime) [reply docs/user/supported_devices.rst:410]: The Archer C50 above is a bad example, sorry for the nitpick.

**Review by AiyionPrime — CHANGES_REQUESTED** [review]
Looking at the 5GHz problems with rev4 I'd recommend testing 5GHz mesh and client network work in parallel. Leaving those two tests out feels like pushing our luck.

Find a proper mesh partner and leave them running for a day or so without crashing interfaces while a client is connected to the device and I'm fine with merging this.

File: docs/user/supported_devices.rst (hunk 1 above)
Comment (AiyionPrime) [reply docs/user/supported_devices.rst:410]: I proposed to take care of the C50 in #2704, no need to add it to your PR :)

**Comment by AiyionPrime** [comment]
Files do look fine. Let us know, when the checklist is completed.

> Looking at the 5GHz problems with rev4 I'd recommend testing 5GHz mesh and client network work in parallel. Leaving those two tests out feels like pushing our luck.
> 
> Find a proper mesh partner and leave them running for a day or so without crashing interfaces while a client is connected to the device and I'm fine with merging this.

**Review by AiyionPrime — APPROVED** [review]

**Comment by misanthropos** [comment]
> > Find a proper mesh partner and leave them running for a day or so without crashing interfaces while a client is connected to the device and I'm fine with merging this.

Done and dusted. No problems so far.

**Pull request merged**